### DISCUSSION TITLE
[1 febbraio 2022 - B] Esercizio 6

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#216] [Esami]
**Esercizio 6 (5 punti):**

Si consideri il numero esadecimale $2A5B$ e gli si sottragga in base 16 il numero esadecimale $9C7$. Si converta poi il risultato in una sequenza binaria di 16 bit, da interpretarsi come un numero razionale in formato IEEE half-precision. Al numero così ottenuto si sottragga il numero con $1010011100000000_2$ espresso sempre in formato IEEE half-precision, e si scriva il risultato nello stesso formato.

**Comment by Elia-Belli** [comment]
- $X = 2A5B - 9C7 = 2094_{16}$
  $X = 0\ 01000\ 0010010100 = +2^{-7}\times 1.0010010100 = +2^{-6}\times 0.10010010100$
  $Y = 1\ 01001\ 1100000000 = -2^{-6}\times 1.11$
- $Z = X-Y$ è la somma di 2 numeri positivi, quindi basta aggiungere un bit per evitare l'overflow
  $00.100100101 +$
  $01.110000000 =$
  $10.010100101 = + 2^{-6}\times 10.010100101 = 2^{-5}\times 1.0010100101$
- $Z = 0\ 01010\ 0010100101 = 28A5_{16}$

**Comment by fraxb04** [comment]
@Elia-Belli corretto
